### PR TITLE
UX-2089/Added workaround for retrieval of cluster addon versions for …

### PR DESCRIPTION
**Pull request by kevinylee1991**
…EKS clusters and fixed issue where addons did not appear on EKS cluster creation wizard

### CODE REVIEW

File: src/app/plugins/infrastructure/components/clusters/aws/capi/CapiAddonsWizardStep.tsx [hunk 1]
```diff
@@ -19,7 +19,25 @@ import metricsServerJson from '../../cluster-addons/schema/metrics-server.json'
...
+  clusterAddonDisplayNames,
+  sameMajorAndMinorVersion,
+  getPatchVersion,
+} from '../../cluster-addons/helpers'
+import { compose, prop, sortBy } from 'ramda'
+
+const sortByPatchVersion = sortBy(compose(getPatchVersion, prop('version')))
+
+// For EKS clusters, the kubernetes version doesn't match the ClusterVersion kubernetes version
+// so we match the major & minor versions, and then take the ClusterVersion with newest patch version
```
Comment (patil-pratik-87) [inline src/app/plugins/infrastructure/components/clusters/aws/capi/CapiAddonsWizardStep.tsx:32]: 👍🏽

File: src/app/plugins/infrastructure/components/clusters/aws/capi/CapiAddonsWizardStep.tsx [hunk 2]
```diff
@@ -62,18 +80,24 @@ const nonConfigurableAddons: AddonManagerAddon[] = [
...
   const { message, loading, reload } = useListAction(listClusterVersions)
   const clusterVersions = useSelector(clusterVersionsSelector)
 
   // Set addon versions
   useEffect(() => {
-    if (wizardContext.version) {
-      const addOnVersions = clusterVersions.find(
-        (clusterVersion) => clusterVersion.version === wizardContext.version,
-      )?.addons
+    const version = wizardContext.version || wizardContext.wgVersion
```
Comment (patil-pratik-87) [inline src/app/plugins/infrastructure/components/clusters/aws/capi/CapiAddonsWizardStep.tsx:96]: Do we need the `wizardContext.wgVersion` here? It should ideally only be used for worker nodes like MachineDeployments and MachinePools.

File: src/app/plugins/infrastructure/components/clusters/aws/capi/CapiAddonsWizardStep.tsx [hunk 3]
```diff
@@ -91,7 +115,7 @@ export default function CapiAddonsWizardStep({ wizardContext, setWizardContext,
         capiAutoScalerVersion: versionMap['cascapi'] || '1.21.1',
       })
     }
-  }, [clusterVersions, wizardContext.version])
+  }, [clusterVersions, wizardContext?.version, wizardContext?.wgVersion])
```
Comment (patil-pratik-87) [inline src/app/plugins/infrastructure/components/clusters/aws/capi/CapiAddonsWizardStep.tsx:118]: same comment as above

**Review by patil-pratik-87 — APPROVED** [review]

File: src/app/plugins/infrastructure/components/clusters/aws/capi/CapiAddonsWizardStep.tsx (hunk 2 above)
Comment (kevinylee1991) [reply src/app/plugins/infrastructure/components/clusters/aws/capi/CapiAddonsWizardStep.tsx:96]: I can change this, but I'll need to change the control plane step for EKS

**Review by patil-pratik-87 — APPROVED** [review]

**Pull request merged**